### PR TITLE
Shrink Docker image size

### PR DESCRIPTION
**Pull request by mboersma**
Using nginx and alpine linux is even smaller than our ubuntu + shell script version:
```console
$ docker images | grep example-
example-dockerfile-http      alpine     2b351866faf8        17 seconds ago      6.491 MB
example-dockerfile-http      ubuntu     c8303c15036b        21 hours ago        87.77 MB
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,7 @@
...
-  apt-get clean && \
-  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+FROM alpine:3.1
 
+RUN apk add -U nginx && mkdir -p /tmp/nginx && rm -rf /var/cache/apk/*
 ENV POWERED_BY Deis
-CMD while true; do echo "HTTP/1.1 200 OK\n\nPowered by $POWERED_BY" | nc -l -p 1500; done
+CMD /usr/local/bin/start.sh
 EXPOSE 1500
+ADD files/ /
```
Comment (bacongobbler) [inline Dockerfile]: small nit: common practice seems to call this `rootfs`

**Comment by bacongobbler** [comment]
LGTM! There's no way we can `nc` in alpine? That would make this image even smaller!

**Comment by mboersma** [comment]
I tried the 'nc` approach but it wouldn't work reliably on alpine for some reason. `nc` there doesn't know the `-p` flag, and after tweaking the flags I could only get it to serve once, not sure exactly why.

File: Dockerfile (hunk 1 above)
Comment (mboersma) [inline Dockerfile]: Sounds good, I'll change it.

**Comment by aledbf** [comment]
@mboersma that's the reason why I close https://github.com/deis/example-dockerfile-http/pull/2

**Comment by bacongobbler** [comment]
> after tweaking the flags I could only get it to serve once, not sure exactly why

That's what the `while true;` loop is for. it listens for only one request before closing the connection, which was weird to me as well. Either case, this is much smaller than the previous implementation (which is what this image was shooting for).

Let me just test that `deis run` still works with this app because that's why we had to shift away from busybox with the original image (since `deis run` couldn't work, the integration tests failed with this app).

**Comment by bacongobbler** [comment]
Yeah, this will fail unless `bash` is installed:

```
docker run --entrypoint=/bin/bash 77f1f3f789d7 env
exec: "/bin/bash": stat /bin/bash: no such file or directoryFATA[0000] Error response from daemon: Cannot start container 33a62436f9f5a8c075d26129da5ca38857de1a5e21bb506ca7a4d6ec188f26be: exec: "/bin/bash": stat /bin/bash: no such file or directory
```

If I modify the `apk add` line to also install `bash`, it works fine.

**Comment by mboersma** [comment]
> If I modify the `apk add` line to also install `bash`, it works fine.

That makes the Docker image balloon up to 7.991MB... I guess that's acceptable, I'll change it. Thanks @aledbf and @bacongobbler.

**Comment by mboersma** [comment]
Closing this in favor of #5, which is much less disruptive and simpler.

**Pull request closed without merging**